xray/core/alignment.py

```diff
@@ -276,14 +325,47 @@ def broadcast_arrays(*args):
             raise ValueError('cannot broadcast arrays: the %s index is not '
                              'aligned (use xray.align first)' % k)
 
-    vars = broadcast_variables(*[a.variable for a in args])
-    indexes = dict((k, all_indexes[k][0]) for k in vars[0].dims)
+    common_coords = OrderedDict()
+    dims_map = OrderedDict()
+    for arg in args:
+        for dim in arg.dims:
+            if dim not in common_coords:
+                common_coords[dim] = arg.coords[dim].variable
+                dims_map[dim] = common_coords[dim].size
+
+    def _broadcast_array(array):
+        data = array.variable.expand_dims(dims_map)
+        coords = OrderedDict(array.coords)
+        coords.update(common_coords)
+        dims = tuple(common_coords)
+        name = array.name
+        attrs = array.attrs
+        encoding = array.encoding
+        return DataArray(data, coords, dims, name, attrs, encoding)
```

Comment:
unpacking all the attrs here seems unnecessarily verbose, not wrong but just more verbose than I'm used to seeing.